### PR TITLE
Big decimal double constructor

### PR DESCRIPTION
**Pull request by cpfeiffer**
See https://github.com/spotbugs/spotbugs/pull/696
Rebased on master
----
- [x ] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Currently the versioning policy of this project follows [Semantic Versioning v2.
 
 ## Unreleased - 2019-??-??
 * Start migrating STDOUT/STDERR usage to a logging framework
+* DMI_BIGDECIMAL_CONSTRUCTED_FROM_DOUBLE was only reported when the double argument was a constant
```
Comment (KengoTODA) [inline CHANGELOG.md]: Is it bug fix? Then please add this entry under `### Fixed`.

**Comment by KengoTODA** [comment]
@cpfeiffer could you merge (or rebase to) the latest `origin/master`?

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue696Test.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Contributions to SpotBugs
+ * Copyright (C) 2018, cpfeiffer
```
Comment (ThrawnCA) [inline spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue696Test.java]: 2019?

File: spotbugsTestCases/src/java/ghIssues/Issue696.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package ghIssues;
+
+import java.math.BigDecimal;
+
+/**
+ * @see <a href="https://github.com/spotbugs/spotbugs/issues/696">GitHub issue</a>
```
Comment (ThrawnCA) [inline spotbugsTestCases/src/java/ghIssues/Issue696.java]: Can there be at least a minimal description here independently of Github?

**Comment by KengoTODA** [comment]
This conflict hasn't been closed for a while, so I will close this PR. Please reopen it freely :)

**Pull request closed without merging**